### PR TITLE
Refactoring AmazonInfo refresh into common class

### PR DESCRIPTION
**Pull request by qiangdavidliu**
+ other minor clean ups

### CODE REVIEW

File: eureka-client/src/main/java/com/netflix/appinfo/RefreshableAmazonInfoHolder.java [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * A holder class for AmazonInfo that exposes some APIs to allow for refreshes.
+ */
+public class RefreshableAmazonInfoHolder {
```
Comment (tcellucci) [inline eureka-client/src/main/java/com/netflix/appinfo/RefreshableAmazonInfoHolder.java]: seems like this class should implement standard interface like Supplier or Provider, for better interop with java api

File: eureka-client/src/main/java/com/netflix/appinfo/RefreshableAmazonInfoHolder.java [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+     */
+    public AmazonInfo getCurrent() {
+        return info;
+    }
+
+
+    /**
+     * Rules of updating AmazonInfo:
+     * - instanceId must exist
+     * - localIp/privateIp most exist
```
Comment (tcellucci) [inline eureka-client/src/main/java/com/netflix/appinfo/RefreshableAmazonInfoHolder.java]: must

File: eureka-client/src/main/java/com/netflix/discovery/shared/transport/decorator/RetryableEurekaHttpClient.java [hunk 3]
```diff
@@ -126,7 +126,7 @@ public void shutdown() {
                 }
                 logger.warn("Request execution failure with status code {}; retrying on another server if available", response.getStatusCode());
             } catch (Exception e) {
-                logger.warn("Request execution failure", e.getMessage());  // just log message as the underlying client should log the stacktrace
+                logger.warn("Request execution failured with message: {}", e.getMessage());  // just log message as the underlying client should log the stacktrace
```
Comment (tcellucci) [inline eureka-client/src/main/java/com/netflix/discovery/shared/transport/decorator/RetryableEurekaHttpClient.java]: failed

**Comment by tcellucci** [comment]
name AmazonInfo is overly broad; they (amazon guys) call it InstanceMetadata in the documentation and I think that's on-target.

File: eureka-client/src/main/java/com/netflix/appinfo/RefreshableAmazonInfoHolder.java (hunk 1 above)
Comment (qiangdavidliu) [inline eureka-client/src/main/java/com/netflix/appinfo/RefreshableAmazonInfoHolder.java]: Yeah thought about that too, I'll stick with Provider for now.

File: eureka-client/src/main/java/com/netflix/discovery/shared/transport/decorator/RetryableEurekaHttpClient.java (hunk 3 above)
Comment (qiangdavidliu) [inline eureka-client/src/main/java/com/netflix/discovery/shared/transport/decorator/RetryableEurekaHttpClient.java]: lol got too lazy on this one:)

**Comment by qiangdavidliu** [comment]
yeah that's stuck forever unfortunately.

**Comment by spencergibb** [comment]
@qiangdavidliu thanks for jumping on this so quickly!

**Pull request merged**